goutubedl.go

```diff
@@ -453,6 +465,23 @@ func infoFromURL(ctx context.Context, rawURL string, options Options) (info Info
 		info.Entries = filteredEntrise
 	}
 
+	// channels contain playlists, so recurse into them
+	if options.Type == TypeChannel {
+		var filteredEntrise []Info
```

Comment:
I guess this is the only resonable flattening order?

Reply:
I deduplicated the 2 code blocks and add some documentation: 9122483.

We could consider using a map, to deduplicate entries that are found in multiple playlists. But I'll probably need to find an example of a (small) channel that does this first.

Full recursion might also be nice to have, but I'll just assume it's overkill until someone comes across a site that requires it.

Reply:
In hindsight, having squashed the two `Entries` collection loops into a single block, a dedicated `TypeChannel` might not really be necessary. But maybe it's good to have around in case we decide to handle channels differently at some point?

Reply:
Nice comment 👍  yeap keep it simple for now sounds good